idx/cassandra/cassandra.go

```diff
@@ -49,6 +51,12 @@ var (
 	// metric idx.cassandra.save.skipped is how many saves have been skipped due to the writeQueue being full
 	statSaveSkipped = stats.NewCounter32("idx.cassandra.save.skipped")
 	errmetrics      = cassandra.NewErrMetrics("idx.cassandra")
+
+	metaRecordRetryPolicy = gocql.ExponentialBackoffRetryPolicy{
```

Comment:
This retry policy seems a bit aggressive.  it will retry 10 times in less then 4seconds.

Reply:
What do you think would make more sense?

Reply:
What failure modes are the retries there to defend against?

Reply:
network issues, the duration of those is not really predictable. if a cassandra pod restarts it can take over a minute for it to come back, so i guess the max backoff should be like 2 min then?

Reply:
if a cassandra pod restarts, the request will just be retried on another pod.

I guess the real question is: how long can you afford to wait? 
If you are protecting against network issues, then attempts will fail pretty quickly if the network is down.  So you can work out the min and max that will allow retries up until your max desired execution time.  eg, if these requests are originating from HTTP requests, then you could wait up to 60seconds (but probably best to target 30s or 45s).

Reply:
Good point, it should definitely be under the HTTP request timeout. These requests are originating from an HTTP request, so in case of a final error we want to return an error to the HTTP client.

Reply:
i explained the calculations in the comment:
https://github.com/grafana/metrictank/pull/1471/commits/88c309bed5415cb3d7ac4680e698bc915602f54c